packages/dx-grid-core/src/plugins/sorting-state/helpers.ts

```diff
@@ -15,7 +15,8 @@ export const getColumnSortingDirection: GetColumnSortingDirectionFn = (
 export const getPersistentSortedColumns: GetPersistentSortedColumnsFn = (
   sorting, columnExtensions = [],
 ) => columnExtensions.reduce((acc, { columnName, sortingEnabled }) => {
-  if (!sortingEnabled) {
+  // tslint:disable-next-line:no-boolean-literal-compare
+  if (sortingEnabled === false) {
```

Comment:
maybe we disable it not only for this line but where the strictNullChecks are false
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unnecessary-boolean-literal-compare.md#not-to-use-it